src/main/java/ch/njol/skript/expressions/ExprItems.java

```diff
@@ -99,59 +101,14 @@ protected ItemStack[] get(Event event) {
 	@Nullable
 	public Iterator<ItemStack> iterator(Event event) {
 		if (!items && itemTypeExpr == null)
-			return new Iterator<ItemStack>() {
+			return new ArrayIterator<>(ALL_BLOCKS.clone());
```

Comment:
I don't think this clones the itemstacks themselves, but rather creates a new array with the same objects. so any modifications would still modify the itemstacks in the original array (which we don't want I think).

Reply:
oh, you're right, how would i fix that?

Reply:
I think you could do `return Arrays.stream(ALL_BLOCKS).map(ItemStack::clone).iterator()`.